foundry/foundry.py

```diff
@@ -820,3 +827,53 @@ def is_doi(string: str):
         return True
     else:
         return False
+def _get_files(tc, ep, queue, max_depth):
+    while queue:
+        abs_path, rel_path, depth = queue.pop()
+        path_prefix = rel_path + "/" if rel_path else ""
+
+        res = tc.operation_ls(ep, path=abs_path)
+
+        if depth < max_depth:
+            queue.extend(
+                (
+                    res["path"] + item["name"],
+                    path_prefix + item["name"],
+                    depth + 1,
+                )
+                for item in res["DATA"]
+                if item["type"] == "dir"
+            )
+        for item in res["DATA"]:
+            if item["type"]=='file':
+                item["name"] = path_prefix + item["name"]
+                item["path"] = abs_path.replace('/~/','/')
+                yield item
+
+def recursive_ls(tc, ep, path, max_depth=3):
+    queue = deque()
+    queue.append((path, "", 0))
+    yield from _get_files(tc, ep, queue, max_depth)
+
+def download_file(item, https_config):
```

Comment:
Could you add a docstring and/or type hints? E.g., what content does "item" have and where does it usually come from? 

Additionally, the indentation is odd here. Does your editor have an autopep you could run on this?